assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -65,32 +105,49 @@ export const actions = {
 		};
 	},
 
-	receiveAccountsPropertiesProfilesCompleted() {
-		return {
-			payload: {},
-			type: RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_COMPLETED,
-		};
+	*resetAccounts() {
+		const registry = yield Data.commonActions.getRegistry();
+
+		yield { type: RESET_ACCOUNTS };
+
+		return registry.stores[ STORE_NAME ].getActions()
+			.invalidateResolutionForStoreSelector( 'getAccounts' );
 	},
 
-	receiveAccountsPropertiesProfilesFailed( error ) {
-		invariant( error, 'error is required.' );
+	*selectAccount( accountID ) {
+		invariant( isValidAccountSelection( accountID ), 'A valid accountID is required to select.' );
 
-		return {
-			payload: { error },
-			type: RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_FAILED,
-		};
+		const registry = yield Data.commonActions.getRegistry();
+		registry.dispatch( STORE_NAME ).setAccountID( accountID );
+		registry.dispatch( STORE_NAME ).setPropertyID( '' );
```

Comment:
I'd be more comfortable keeping the empty string assignments here. While it may feel weird, PHP expects these values to be strings. I'm not entirely sure how good our sanitization there is, so I'd prefer to keep values (particularly those that interact with the server) true to their expected data types.

I personally overall prefer using `null` only for objects. The empty string is typically the default uninitialized value for string-type values, so I think specifying that here makes sense.

Reply:
An empty string is consistent with what the default option value is on the server side.

Reply:
😅 Fair enough!